tests/config/test_config_comparator.py

```diff
@@ -54,41 +57,119 @@ def test_config_comparator():
+    assert blocked_config_diff["added_items"][1] == (
         ("section_name", "section_4", None),
         {"attribute": "4:int", "prop": "prop_4"},
     )
+    assert blocked_config_diff["added_items"][0] == (
+        ("Global Configuration", "repository_properties", None),
+        {"foo": "bar"},
+    )
 
-    # TODO: This should not in ["added_items"] since it is a default value that was changed. This should be fixed soon.
-    assert config_diff["added_items"][0] == (("Global Configuration", "repository_properties", None), {"foo": "bar"})
-
-    assert len(config_diff["modified_items"]) == 8
-    assert config_diff["modified_items"][0] == (("Global Configuration", "root_folder", None), ("./taipy/", "foo"))
-    assert config_diff["modified_items"][1] == (("Global Configuration", "storage_folder", None), (".data/", "bar"))
-    assert config_diff["modified_items"][2] == (
+    assert len(blocked_config_diff["modified_items"]) == 8
+    assert blocked_config_diff["modified_items"][0] == (
+        ("Global Configuration", "root_folder", None),
+        ("./taipy/", "foo"),
+    )
+    assert blocked_config_diff["modified_items"][1] == (
+        ("Global Configuration", "storage_folder", None),
+        (".data/", "bar"),
+    )
+    assert blocked_config_diff["modified_items"][2] == (
         ("Global Configuration", "clean_entities_enabled", None),
         ("ENV[TAIPY_CLEAN_ENTITIES_ENABLED]", "True:bool"),
     )
-    assert config_diff["modified_items"][3] == (
+    assert blocked_config_diff["modified_items"][3] == (
         ("Global Configuration", "repository_type", None),
         ("filesystem", "baz"),
     )
-    assert config_diff["modified_items"][4] == (("section_name", "section_2", "attribute"), ("2:int", "20:int"))
-    assert config_diff["modified_items"][5] == (("section_name", "section_2", "prop"), ("prop_2", "prop_2b"))
-    assert config_diff["modified_items"][6] == (
+    assert blocked_config_diff["modified_items"][4] == (("section_name", "section_2", "attribute"), ("2:int", "20:int"))
+    assert blocked_config_diff["modified_items"][5] == (("section_name", "section_2", "prop"), ("prop_2", "prop_2b"))
+    assert blocked_config_diff["modified_items"][6] == (
         ("section_name", "section_5", "prop"),
         (["prop_5"], ["prop_4", "prop_5"]),
     )
-    assert config_diff["modified_items"][7] == (
+    assert blocked_config_diff["modified_items"][7] == (
         ("section_name", "section_5", "attribute"),
         (["1:int", "2:int", "3:int", "4:int", "5:int"], ["1:int", "2:int"]),
     )
 
-    assert len(config_diff["removed_items"]) == 1
-    assert config_diff["removed_items"][0] == (
+    assert len(blocked_config_diff["removed_items"]) == 1
+    assert blocked_config_diff["removed_items"][0] == (
         ("section_name", "section_3", None),
         {"attribute": "3:int", "prop": "prop_3"},
     )
+
+
+def test_unblock_section_when_compare(caplog):
```

Comment:
I would split this test as well. What about
- one test to test the _unblock_section is modifying the `_UNBLOCKED_SECTIONS` variable
- one test to assert non-blocking diffs are collected in a separate variable
- one test to check the logging is correct